Tesla bug fixes + new component.

zabuldon opened the pull request:
## Description:
Fixed issues wit component.
Added new Switch to control Tesla charging.

**Related issue (if applicable):** fixes #9381 #9370

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#3404

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

arsaboo committed:
Fixes #9379 - Added additional string check in Wunderground sensor (#9380)

* Added additional string check

* optimaze

MikeChristianson committed:
Fixes #9353 (#9354)

Follow [Twitter's guidance](https://dev.twitter.com/rest/reference/post/media/upload-finalize) for media uploads: "If and (only if) the response of the FINALIZE command contains a processing_info field, it may also be necessary to use a STATUS command and wait for it to return success before proceeding to Tweet creation."

viswa-swami committed:
Fixing foscam library dependency/requirements (#9387)

* Added support to enable/disable motion detection for foscam cameras. This support was added in 0.48.1 as a generic service for cameras. Motion detection can be enabled/disabled for foscam cameras with this code-set.

* Fixed the violation identified by hound-bot

* Fixed the comment posted by HoundCI-Bot regarding using imperative mood statement for pydocstyle

* Fixed the error that travis-ci bot found.

* As per comment from @balloob, Instead of directly using the URL to talk to foscam, used a 3rd party foscam library to communicate with it. This library already has support to enable/disable motion detection and also APIs to change the motion detection schedule etc. Need to add more support in the pyfoscam 3rd party library for checking if motion was detected or even if sound was detected. Once that is done, we can add that into HASS as well.

* Lint

* Removed the requests library import which is not used anymore

* Updating requirements_all.txt based on the code-base of home assistant that i have. Generated using the gen_requirements_all.py script

* Updating requirements_all.txt and requirements_test_all.txt generated by gen_requirements_all.py after latest pull from origin/dev

* Updated requirements_all.txt with script

* Updated the foscam camera code to fix lint errors

* Fixed houndci violation

* Updating the foscam library dependency/requirements.

* Fixing the requirements_all file. Somehow when i generated, it generated duplicate entry for the same dependency

balloob committed:
Update frontend

balloob committed:
Version bump to 0.53.1

balloob committed:
Merge pull request #9395 from home-assistant/release-0-53-1

0.53.1

zabuldon committed:
Fix for #9381 and #9370.
Extended logging enabled.

zabuldon committed:
Fix for #9381 and #9370.
Extended logging enabled.

zabuldon committed:
Renamed TeslaChargerSwitch to ChargerSwitch

zabuldon committed:
Renamed TeslaChargerSwitch to ChargerSwitch

zabuldon committed:
lint fix

zabuldon committed:
lint fix

Julius2342 committed:
renamed add_devices to async_add_devices according to hass naming scheme (#9485)

* renamed add_devices to async_add_devices according to hass naming scheme

* replaced some occurencies of async_add_entites to async_add_devices

* fixed unit test

* fixed unit test

pvizeli commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,58 @@
+        """Initialisation of the switch."""
+        self._state = None
+        super().__init__(tesla_device, controller)
+        self._name = self.tesla_device.name
```
will be done with TeslaDevice object

pvizeli commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,58 @@
+        self._state = None
+        super().__init__(tesla_device, controller)
+        self._name = self.tesla_device.name
+        self.entity_id = ENTITY_ID_FORMAT.format(self.tesla_id)
```
That is not needed...

zabuldon replied:
Can we discuss that in discord? There is some questions from my side.

balloob replied:
This is optional. If you have a lot of Teslas with overlapping names, having it using the tesla ID is a good idea.

zabuldon replied:
So, Tesla component allow you to integrate a few Tesla's. It is why I'm using VIN number as a part of the entity_id. 
Could you please confirm if I can continue to use assignment of entity_id in my code?

balloob replied:
Yes, you can keep using the vin.

pvizeli reviewed (changes requested):
Please remove all this debug stuff. You can see all that Infos with `info` level on service calls. It will slow done the core since all this message going trought the event Loop. So all message that realy not nessasary should be cleanup for prod version

pvizeli committed:
Revert "renamed add_devices to async_add_devices according to hass naming scheme (#9485)" (#9503)

This reverts commit a5a970709f02b5ef0be96af434371806863f81c7.

amelchio committed:
LIFX: fix multi-zone color restore after effects (#9492)

The aiolifx 0.6.0 release fixes an issue where an effect color could
remain set after stopping the effect. This only affected multi-zone
lights (i.e. LIFX Z) and only if the effect was stopped by setting
the light brightness or the color (but not both).

The aiolifx 0.6.0 release also defaults end_index to start_index+7,
so we can remove that argument.

Finally, aiolifx_effects 0.1.2 adds support for aiolifx 0.6.0.

Julius2342 committed:
renamed add_devices to async_add_devices according to hass naming scheme (second try after failed #9485) (#9505)

Danielhiversen committed:
Xiaomi pycryptodome (#9511)

* Switch to use pycryptodome for xiaomi_gw

danielperna84 committed:
Bumped pyhomematic, additional device support (#9506)

Add an optional extended description…

vickyg3 committed:
abode: Bump abodepy dependency to 0.11.7 (#9504)

* abode: Bump abodepy dependency to 0.11.7

Fixes cases where one's abode account has a nest thermostat
linked (https://github.com/MisterWil/abodepy/pull/17).

* abode: Bump abodepy dependency to 0.11.7

Fixes cases where one's abode account has a nest thermostat
linked (https://github.com/MisterWil/abodepy/pull/17).

* update requirements_all.txt

vickyg3 committed:
abode: Set device_type in state attributes (#9515)

This gets displayed when clicking on the binary sensors. It is
useful to distinguish different devices with the same name (e.g.
the room name) but different types.

Danielhiversen committed:
update xiaomi aqara lib (#9520)

skalavala committed:
added services.yaml integration for input_boolean (#9519)

* added services.yaml integration to input_boolean

* added services integration for input_boolean

* removed trailing spaces

doudz committed:
Add reload service to python_script (#9512)

* Add reload service

* add reload test

* Use global variable

* remove white space ....

* adjust as suggested

* remove annoying white space....

* fix travis

* fix travis, again

* rename Load_scripts to Discover_scripts

Travis complains that "Load_scripts" is an invalid name (I don't know why)

* Update python_script.py

marthoc committed:
MQTT Binary Sensor - Add availability_topic for online/offline status (#9507)

* MQTT Binary Sensor - Add availability_topic for online/offline status

Added topic, configurable payloads, and tests.

* Relocated state subscribe function

Moved state subscribe function to follow the state listener function.

Julius2342 committed:
Fix typo within cover/knx https://github.com/XKNX/xknx/issues/64 (#9527)

amelchio committed:
LIFX: improve performance of setting multi-zone lights to a single color (#9526)

With this optimization we can send a single UDP packet to the light rather
than one packet per zone (up to 80 packets for LIFX Z). This removes a
potential multi-second latency on the frontend color picker.

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Update frontend

balloob committed:
Version bump to 0.54

balloob committed:
Version bump to 0.55.0.dev0

Danielhiversen committed:
flux led lib 0.20 (#9533)

MisterWil committed:
Update AbodePy to 0.11.8 (#9537)

* Update requirements_all.txt

* Update abode.py

MisterWil committed:
Update AbodePy to 0.11.8 (#9537)

* Update requirements_all.txt

* Update abode.py

arsaboo committed:
Bump python_openzwave to 0.4.0.35 (#9542)

* Bump python_openzwave to 0.4.0.35

* Cleanup

arsaboo committed:
Bump python_openzwave to 0.4.0.35 (#9542)

* Bump python_openzwave to 0.4.0.35

* Cleanup

happyleavesaoc committed:
update usps (#9540)

* update usps

* fix syntax issue

happyleavesaoc committed:
update usps (#9540)

* update usps

* fix syntax issue

balloob committed:
Merge pull request #9532 from home-assistant/release-0-54

0.54

pvizeli committed:
Bugfix Homematic hub object (#9544)

* Bugfix Homematic hub object

* fix hass instance

* fix state unknow if 0 states

mika committed:
Fix a bunch of typos (#9545)

s/Addres /Address /
s/Chnage/Change/
s/Converion/Conversion/
s/Supressing/Suppressing/
s/agains /against /
s/allready/already/
s/analagous/analogous/
s/aquired/acquired/
s/arbitray/arbitrary/
s/argment/argument/
s/aroung/around/
s/attibute/attribute/
s/auxillary/auxiliary/
s/befor /before /
s/commmand/command/
s/conatin/contain/
s/conection/connection/
s/coresponding/corresponding/
s/entites/entities/
s/enviroment/environment/
s/everyhing/everything/
s/expected expected/expected/
s/explicity/explicitly/
s/formated/formatted/
s/incomming/incoming/
s/informations/information/
s/inital/initial/
s/inteface/interface/
s/interupt/interrupt/
s/mimick/mimic/
s/mulitple/multiple/
s/multible/multiple/
s/occured/occurred/
s/occuring/occurring/
s/overrided/overridden/
s/overriden/overridden/
s/platfrom/platform/
s/positon/position/
s/progess/progress/
s/recieved/received/
s/reciever/receiver/
s/recieving/receiving/
s/reponse/response/
s/representaion/representation/
s/resgister/register/
s/retrive/retrieve/
s/reuqests/requests/
s/segements/segments/
s/seperated/separated/
s/sheduled/scheduled/
s/succesfully/successfully/
s/suppport/support/
s/targetting/targeting/
s/thats/that's/
s/the the/the/
s/unkown/unknown/
s/verison/version/
s/while loggin out/while logging out/

Tommatheussen committed:
Catch no longer existing process in systemmonitor (#9535)

* Catch no longer existing process in systemmonitor

* Update log message

* Again line length

fabaff committed:
Upgrade pyasn1 to 0.3.6 (#9548)

andrey-git committed:
Add history_graph component (#9472)

* Add support for multi-entity recent fetch of history. Add graph component

* Rename graph to history_graph. Support fast fetch without current state.

* Address comments

exxamalte committed:
GeoRSS sensor (#9331)

* new geo rss events sensor

* SCAN_INTERVAL instead of DEFAULT_SCAN_INTERVAL

* removed redefinition CONF_SCAN_INTERVAL

* definition of self._name not required

* removed unnecessary check and unnecessary parameter

* changed log levels

* fixed default name not used

* streamlined sensor name and entity id generation, removed unnecessary parameter

* fixed issue for entries without geometry data

* fixed tests after code changes

* simplified code

* simplified code; removed unnecessary imports

* fixed invalid variable name

* shorter sensor name and in turn entity id

* increasing test coverage for previously untested code

* fixed indentation and variable usage

* simplified test code

* merged two similar tests

* fixed an issue if no data could be fetched from external service; added test case for this case

MisterWil committed:
Fixed bug with all switch devices being excluded (#9555)

happyleavesaoc committed:
fix usps? (#9557)

snjoetw committed:
Added support for ARM_NIGHT for manual_mqtt alarm (#9358)

* - Added support for ARM_NIGHT for manual_mqtt alarm

* - port "Add post_pending_state attribute to manual alarm_control_panel #9291" to manuql_mqtt

* - port "Fixed manual alarm not re-arm after 2nd trigger #9249" to manuql_mqtt

* - port "Add manual alarm_control_panel pending time per state #9264" to manuql_mqtt

* - Updated test_trigger_with_specific_pending to simulate real scenario e.g. arm the system then trigger

zabuldon commented:
yes. my component allows you to control >1 of cars. it is why VIN number is a part of entity id. 

Do I still need to remove it if no, I'll remove debug and it should be ready to merge.

bachya committed:
Various AirVisual bugfixes (#9554)

* Various AirVisual bugfixes

* Updating requirements

* Added better logging for failed data retrieval

bachya committed:
Updated Arlo cameras with new attributes (#9565)

balloob committed:
Allow specifying multiple ports for UPNP component (#9560)

* Update UPNP component

* Bump dep

* Fix flakiness in test

pfalcon committed:
Update yeelight to 0.3.3. (#9561)

Fixes basic light control in case complex transition effects were defined
on a light (possibly, externally to Home Assistant):
https://gitlab.com/stavros/python-yeelight/issues/17

aequitas committed:
Add test cases and fix for device_defaults fire_event option. (#9567)

* Add test cases and fix for device_defaults fire_event option.

* Also for light.

* Change docstring mood.

fabaff committed:
Use simplepush module, enable event, and allow encrypted communication (#9568)

* Use simplepush module, enable event, and allow encrypted communication

* Fix check

balloob committed:
Add DuckDNS component (#9556)

* Add DuckDNS component

* Address comments

balloob committed:
Clean up OwnTracks (#9569)

* Clean up OwnTracks

* Address comments

marthoc committed:
MQTT Cover: Add availability topic and configurable payloads (#9445)

* MQTT Cover - Add availability_topic for online/offline status

Added topic, configurable payloads, and tests.

* Merge branch 'dev' into mqtt-cover-availability

* Revert "Merge branch 'dev' into mqtt-cover-availability"

This reverts commit 46d29794ba959e0394ff5c9904ae039a6df1d22e.

* Added newline at end of test_mqtt.py

* Fixed lint issue (newline at EOF)

* Fixed lint issue (newline at EOF)

* Updated call signature for other tests

* Fixed availability message callback

fabaff committed:
Upgrade dsmr_parser to 0.11 (#9576)

sti0 committed:
Added new statistic attributes (#9433)

* Added new statistic attributes

Added new attributes:
  - Cleaning count
  - Total cleaning time
  - Total cleaning area
  - Time left to change main brush, side brush and filter

* Code corrections

Code corrections

* Remove wronge hanging indentation

* Added new attributes

ATTR_MAIN_BRUSH_LEFT
ATTR_SIDE_BRUSH_LEFT
ATTR_FILTER_LEFT
ATTR_CLEANING_COUNT
ATTR_CLEANED_TOTAL_AREA
ATTR_CLEANING_TOTAL_TIME

* Remove trailing white space

* Corrections of the unit test for new attributes

* Hound corrections

* Init self.clean_history, self.consumable_state

* Hound correction

* - Cleaning time and total cleaning time shown in minutes
- Cleaned area and total cleaned area shown in square meters
- Main brush left, side brush left, filter left time shown in hours
- Display of the unit of measurement

* Remove trailing white spaces

* Fixed wrong continued indentation

* Fixed Hound

* Fixed Hound

* Added new statistic attributes

Added new attributes:
  - Cleaning count
  - Total cleaning time
  - Total cleaning area
  - Time left to change main brush, side brush and filter

* Code corrections

Code corrections

* Remove wronge hanging indentation

* Init self.clean_history, self.consumable_state

* Hound correction

* Remove UOM

* Merge

* Init self.clean_history, self.consumable_state

* Hound correction

* Init self.clean_history, self.consumable_state

* Hound correction

* Removed double declarations

fabaff committed:
Upgrade youtube_dl to 2017.9.24 (#9575)

enriquegh committed:
Upgrade lyft_rides to 0.2 (#9578)

joe248 committed:
Increase Comed timeout since it sometimes takes a long time for the API to respond (#9536)

* Increase Comed timeout since it sometimes takes a long time for the API to respond

* Rewrite ComEd sensor to use asyncio

* Fix whitespace and build issues

cmsimike committed:
Allow customizable turn on action for LG WebOS tv (#9206)

* allow customizable action for webos tv turn on as not all models allow for WOL

* trying to fix the houndci-bot

* last few fixes hopefully

* I guess not

* last time!

* This is a breaking change. I have removed the build-in wake-on-lan functionality and have opted for a script which can be a wake-on-lan switch. I have also removed any reference to wol.

* hoping to fix formatting

* linter errors

amelchio committed:
IMAP Unread sensor updated for async and push (#9562)

* IMAP Unread sensor updated for async and push

* Implement renames suggested in review

* Use async_timeout

* Keep push capability in a variable

* Reword for Hound

rbflurry committed:
Replace emulated_hue: with emulated_hue_hidden: for consistency. (#9382)

* Update __init__.py

* fix lint errors

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

Lint errors

* use get_deprecated instead to log old attr

* Updated tests to hide fan.ceiling_fan

* remove space fix lint

fabaff committed:
Upgrade Sphinx to 1.6.4 (#9584)

postlund committed:
Bump pyatv to 0.3.5 (#9586)

Commit:
New Wink services. pair new device, rename, and delete, add new lock key code. Add water heater support (#9303)

* Pair new device, rename, delete, and lock key code services. Also add water heater support.

* Fixed tox

zabuldon committed:
Removed extra assignment of the "name".
Removed unnecessary debug output.
Requirements version bump.

zabuldon committed:
Removed extra assignment of the "name".
Removed unnecessary debug output.
Requirements version bump.

bachya committed:
Fixes UPS MyChoice exception (#9587)

* Fixes UPS MyChoice exception

* Added unit of measurement

* Collaborator-requested changes

bachya committed:
FedEx: Adds "packages" as a unit (#9588)

* Adds "packages" as a unit

* Collaborator-requested changes

pvizeli committed:
Cleanup entity & remove warning (#9606)

* Cleanup entity & remove warning

* Update comment

balloob committed:
Add OwnTracks over HTTP (#9582)

* Add OwnTracks over HTTP

* Fix tests

nkgilley committed:
upgrade python-ecobee-api (#9612)

balloob commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,57 @@
+        """Send the on command."""
+        _LOGGER.debug("Enable charging: %s", self._name)
+        self.tesla_device.start_charge()
+        self._state = STATE_ON
```
We should not set the state inside the lock/unlock functions. Because this device is getting polled, Home Assistant will call the update function after it calls `lock` or `unlock`

djchen committed:
Change TP-Link Switch power statistics attribute names (#9607)

balloob committed:
Migrate Alexa smart home to registry (#9616)

* Migrate Alexa smart home to registry

* Fix tests

tinloaf committed:
Add an input_datetime (#9313)

* Initial proposal for the input_datetime

* Linting

* Further linting, don't define time validation twice

* Make pylint *and* flake8 happy at the same time

* Move todos to the PR to make lint happy

* Actually validate the type of date/time

* First testing

* Linting

* Address code review issues

* Code review: Remove forgotten print()s

* Make set_datetime a coroutine

* Create contains_at_least_one_key_value CV method, use it

* Add timestamp to the attributes

* Test and fix corner case where restore data is bogus

* Add FIXME

* Fix date/time setting

* Fix Validation

* Merge date / time validation, add tests

* Simplify service data validation

* No default for initial state, allow 'unknown' as state

* cleanup

* fix schema

Commit:
Cover component for RFlink (#9432)

* second try on rflink / cover

* no newline at end of file

* changed entity

* fixed comments from pvizeli

* removed :

* removed return 'unknown'

* Fixed comments from Rytilahti

* removed newline

* Reverted to None

* cleanup

* Cleanup

MartinHjelmare commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,57 @@
+        super().__init__(tesla_device, controller)
+        self.entity_id = ENTITY_ID_FORMAT.format(self.tesla_id)
+
+    def lock(self, **kwargs):
```
Shouldn't this be `turn_on`?

MartinHjelmare commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,57 @@
+        self.tesla_device.start_charge()
+        self._state = STATE_ON
+
+    def unlock(self, **kwargs):
```
`turn_off`?

tchellomello committed:
Introducing support to Melnor RainCloud sprinkler systems (#9287)

*  Introducing support to Melnor RainCloud sprinkler systems

* Make monitored_conditions optional for sub-components

*  Part 1/2 - Modified attributes, added DATA_ constant and using battery helper

* Part 2/2 - Refactored self-update hub

* Fixed change requested:
- Dispatcher signal connection
- Don't send raincloud object via dispatcher_send()
- Honoring the dynamic scan_interval value on track_time_interval()

* Inherents async_added_to_hass() on all device classes

* Makes lint happy

* * Refactored RainCloud code to incorporate suggestions.
  Many thanks to @pvizelli and @martinhjelmare!!

* Removed Entity from RainCloud sensor and fixed docstrings

* Update raincloud.py

* Update raincloud.py

* fix lint

snjoetw committed:
Rewrite synology camera by using py-synology package (#9583)

* - Rewrite synology camera by intruducing Api and SurveillanceStation classes to get cameras, motion settings, enable/disable motion detection, etc ...
- Synology camera now shows correct state based on is_recording and is_streaming flag. Also it now supports enable / disable motion detection and show the correct motion detection status
- Newly added Api and SurveillanceStation classes will be moved to a lib but it's here just for review

* - Updated how payload are merged with kwargs so it works with python <3.5

* - Fixed class name conflict

* - Addressed flake8 error

* - Addressed pylint error

* - Moved synology API related code to py-synology lib
- Added py-synology==0.1.1 requirement
- Removed hass from SynologyCamera constructor

* - Updated requirements_all.txt

* - renamed variable back to original

* - Sync call to retrieve camera image should be done in camera_image() instead

* - Sync call to update camera info should be done in update() instead

* - Removed unused import

rytilahti committed:
yeelight: allow turn_off transitions, fixes #9602 (#9605)

fabaff committed:
Move 'voltage' to const (#9621)

jalmeroth committed:
Yamaha MusicCast: check known_hosts (#9580)

* Yamaha MusicCast: check known_hosts

- pymusiccast: Version bump

* Update requirements

etsinko committed:
Fixed away_mode for Ecobee thermostat. (#9559)

* Fixed away_mode for Ecobee thermostat. Now away mode is properly turned on using indefinite away hold.

* fixed lint warnings

* fixed lint warnings

* - now it is possible to use float values for ecobee temperature holds
- fixed a bug that caused an exception when temperature hold was set in away mode
- added unit tests for ecobee thermostat

* fixed lint errors

* fixed lint errors

alanfischer committed:
Switched VeraSensor to use category ids (#9624)

philk committed:
splunk: Handle datetime objects in event payload (#9628)

If an event contained a datetime.datetime object it would cause an
exception in the Splunk component. Most of the media_player
components do this in their `media_position_updated_at` attribute.

Use the JSONEncoder from homeassistant.remote instead of just using the
standard json.dumps encoder.

Fixes #9590

tinloaf committed:
MQTT climate platform [continuation of #8750] (#9589)

* New climate platform with MQTT

* Use STATE_OFF

* Basic tests for climate.mqtt

* lint

* actualy collect coverage

* First tests and fixes

* Add possibility to receive temperature via MQTT

* Require only either sensor or mqtt topic

* Add mqtt publishing for away mode, hold mode and aux heat.

* Use configurabe on/off payloads

* Add pessimistic mode

* Initialize aux and away with False instead of None

* Remove Sensor

* Use correct scheduling method

* Move all methods to coroutines

szogi committed:
wunderground: fix supported language codes #9631 (#9633)

* removed PU, added TR language code (https://www.wunderground.com/weather/api/d/docs?d=language-support&MR=1), fixes #9631

rytilahti committed:
add myself to codeowners (#9642)

szogi committed:
raspihats: unmet dependency fix (#9638)

* raspihats: update to 2.2.3 (deps fix)

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3, smbus-cffi dependency

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3

* raspihats: update to 2.2.3, smbus-cffi dependency

* raspihats: update to 2.2.3, smbus-cffi dependency

* raspihats: update to 2.2.3 (deps fix)

Raspihats platform update, upstream fixed enum34 requirements, added smbus dependency
Fixes #9547

* raspihats: update to 2.2.3, smbus-cffi dependency

David-Byrne committed:
Fixes broken source links in API docs (#9636)

* Fixes broken source links in API docs

* Removes illegal blank line

balloob committed:
Move group services into their own YAML (#9597)

* Move group services into their own YAML

* Fix lint

* Move persistent notification to package

mweimerskirch committed:
Facebook Messenger notify component: add support for sending messages to specific page user IDs (#9643)

vickyg3 committed:
arlo: Add battery level sensor (#9637)

* arlo: Add battery level sensor

Adds a battery level sensor that monitors the battery level on
Arlo cameras.

* Fix lint issue

hotplot committed:
Add hysteresis attribute to threshold binary sensor (#9596)

* Added hysteresis attribute to threshold binary sensor

* Added threshold binary sensor hysteresis test case

* Changed threshold binary sensor property name to be more self explanatory

* Pulled default hysteresis value into top level declaration

* Fixed linter errors

* Fixed additional linter errors

* Move comment to docs

fabaff committed:
Upgrade numpy to 1.13.3 (#9646)

fabaff committed:
Upgrade youtube_dl to 2017.10.01 (#9647)

fabaff committed:
Upgrade discord.py to 0.16.12 (#9648)

fabaff committed:
Upgrade netdisco to 1.2.2 (#9649)

fabaff committed:
Upgrade influxdb to 4.1.1 (#9652)

* Upgrade influxdb to 4.1.1

* Upgrade influxdb to 4.1.1

fabaff committed:
Upgrade googlemaps to 2.5.1 (#9653)

mw-white committed:
mqtt_statestream: Add options to publish attributes/timestamps (#9645)

jeroenterheerdt committed:
Fixed bugs related to exception handling in pythonegardia. Updating package requirement accordingly (#9663)

arsaboo committed:
Update google-api-python-client to 1.6.4 (#9658)

MisterWil committed:
Bump abode to 0.11.9 (#9660)

emlove committed:
Unit tests to improve core coverage (#9659)

* Code coverage of logging util

* Improve async util coverage

* Add test coverage for restore_state

* get_random_string test

alanfischer committed:
Upgrade pyitachip2ir to 0.0.7 (#9669)

zabuldon committed:
Removed extra assignment of components states.
renamed lock/unlock to turn_on/turn_off in switch/tesla.py

zabuldon committed:
Removed extra assignment of components states.
renamed lock/unlock to turn_on/turn_off in switch/tesla.py

FletcherAU committed:
Fix typo in cancel_command description (#9671)

"wasn't going to use it"

MartinHjelmare commented on homeassistant/components/switch/tesla.py:
```diff
@@ -0,0 +1,55 @@
+from homeassistant.const import STATE_ON, STATE_OFF
+
+_LOGGER = logging.getLogger(__name__)
+_VALID_STATES = [STATE_ON, STATE_OFF]
```
Doesn't seem to be used.

BioSehnsucht committed:
Rename input_slider to input_number and add numeric text box option (#9494)

* * Rename input_slider to input_number
* Update input_number to optionally display slider, input box, or both

* input_number support either input box or slider mode, but not both

* input_number : change service from select_value to set_value

* input_number : add test for mode setting to tests

alanfischer committed:
Properly handle an invalid end_time (#9675)

pvizeli committed:
Support new feature to push API data to hassio (#9679)

* Support new featuer to push API data to hassio

* Add tests & services

jeroenterheerdt committed:
Adding ignore capability to Egardia component (#9676)

mbrrg committed:
Fix Google Calendar/oauth2client warning (#9677)

* Fixes oauth2client warning.

* Fix permission.

aequitas committed:
Implement DSMR5 support. (#9686)

* Allow configuring DSMR5 protocol.

* Give good example.

* Using dev branch until released upstream.

* Update to dsmr_parser supporting v5 arguments.

* Update to latest dmsr parser, preventing exceptions thrown where warnings would suffice.

* Update even more

* Update requirements.

* Update requirements

Danielhiversen committed:
Add Tibber sensor (#9661)

* Add Tibber sensor

* remove extra space

milanvo committed:
Add recorder purge service, rework purge timer (#9523)

* Add recorder purge service

* Recorder test to match purge config

* Removed purge timer, move service handler to setup, add service description file

* Tests for recorder purge service

* Recorder purge timer rework, add purge service parameter, tests

* Purge service schema change

* Service description change value range

* First cleanup

* Fix name of config

milanvo committed:
Fix restore state by filter out null value row from DB query (#9690)

jeroenterheerdt committed:
Updating clicksendaudio component based on feedback (#9692)

* Updating clicksendaudio component based on feedback

* Updating .coveragerc - forgot to add new file clicksendaudio.py

bestlibre committed:
Refactoring of onewire sensor component (#9691)

pvizeli committed:
HassIO replace config changes (#9695)

* Update flow

* fix tests

* Update hassio.py

arsaboo committed:
Fixed typo in opencv (#9697)

lwis committed:
[light.tradfri] async support with resource observation. (#7815)

* [light.tradfri] Initial support for observe

* Update for pytradfri 2.0

* Fix imports

* Fix missing call

* Don't yield from add devices

* Fix imports

* Minor fixes to async code.

* Imports, formatting

* Docker updates, some minor async code changes.

* Lint

* Lint

* Update pytradfri

* Minor updates for release version

* Build fixes

* Retry observation if failed

* Revert

* Additional logging, fix returns

* Fix rename

* Bump version

* Bump version

* Support transitions

* Lint

* Fix transitions

* Update Dockerfile

* Set temp first

* Observation error handling

* Lint

* Lint

* Lint

* Merge upstream changes

* Fix bugs

* Fix bugs

* Fix bugs

* Lint

* Add sensor

* Add sensor

* Move sensor attrs

* Filter devices better

* Lint

* Address comments

* Pin aiocoap

* Fix bug if no devices

* Requirements

balloob committed:
RFC: Create a secrets file and enable HTTP password by default (#9685)

* Create a secret and enable password by default

* Comment out api password secret

* Lint/fix tests

fabaff committed:
Support for The Things Network (#9627)

* Support for The Things network's Data Storage

* Rename platform and other changes (async and dict)

* Rename sensor platform and remove check for 200

balloob committed:
Update frontend

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

flowolf committed:
fixed duplicate words (#9705)

happyleavesaoc committed:
move icon battery function from util to helpers (#9708)

balloob committed:
Fix coap commit (#9712)

balloob committed:
Deprecate Python 3.4 support (#9684)

* Deprecate Python 3.4 support

* Update text

tchellomello committed:
Updating helper's icon_for_battery_level location (#9594)

danielperna84 committed:
Upgrade pyhomematic, add path setting and HM-CC-VG-1 support (#9707)

* Bump pyhomematic, add path setting, HM-CC-VG-1 support

* Added requirement

rytilahti committed:
bump the version and catch all exceptions to avoid showing backtraces… (#9720)

* bump the version and catch all exceptions to avoid showing backtraces but a more sane error message

* catch only BTLEExceptions, fix logging strings

fronzbot committed:
More netdata sensors (#9719)

* Added more netdata sensors

* Changed precision on counts, packets, and uptime

fabaff committed:
Upgrade pysnmp to 4.3.10 (#9722)

vickyg3 committed:
arlo: Add alarm control panel component (#9711)

* arlo: Add alarm control panel component

Allows importing arlo base stations as an alarm control panel
component in HA. Lets the users configure a custom home mode since
arlo does not have a built-in home mode.

* fix lint and houndci comments

* Use async_update to update the state

Move the state updating code from state() to update() since it does
I/O.

* Do not set state in __init__

Make sure that update is called by passing the second parameter to
async_add_devices.

* Order imports and fix dos-strings

MisterWil committed:
Abode Temp, Humidity, and Light Sensor (#9709)

* Update to 0.12.1 and sensor implementation.

* Removing unnecessary dict gets.

* Added name property to actually use the _name variable.

* Update docstring

fabaff committed:
Arlo clean-up (#9725)

* Fix remaining isses from #9711

* More clean-up

tchellomello committed:
Introducing support to Travis-CI (#9701)

* Introduced support to Travis CI

* Added Last Build Started sensor and simplified code

* Fixed logic error

* Simplified _LOGGER.debug statement

* Introduced support to Travis CI

* Added Last Build Started sensor and simplified code

* Fixed logic error

* Simplified _LOGGER.debug statement

* Renamed parameter since the repository_names expects a list

* Refactoring code to synchronous

* Simplified variables names

bachya committed:
Add show_on_map config option to AirVisual (#9654)

* Removed lat/long attributes

* Linting

* Revised PR to focus on show_on_map configuration

fabaff committed:
Move 'show_on_map' to const (#9727)

fabaff committed:
Bump release to 0.56.0dev (#9726)

lwis committed:
[light.tradfri] Clone all of aiocoap to ensure pinned commit will be present (#9713)

andrey-git committed:
Add andrey-git to codeowners (#9718)

ryanm101 committed:
Fix: Last Played Media Title persists in plex (#9664)

* Fix: Last Played Media Title in plex would stay even when player was idle/off
     Primary Fix is in the "if self._device" portion.
     code in "if self._session" is a catch all but i'm not 100% if it is needed.

* Fixed lint issues with previous commit

* 1st Pass at refactoring plex refresh
Moved _media** into clearMedia() which is called in _init_ and
at start of refresh.

Removed redunant _media_* = None entries

Grouped TV Show and Music under single if rather than testing
seperately for now.

* Fixed invalid name for _clearMedia()
Removed another media_* = None entry

* Removed print() statements used for debug

* Removed unneeded "if" statement

emlove committed:
Event trigger nested conditions (#9732)

* Test to supported nested event triggers

* Update event trigger to allow nested data tests

pvizeli committed:
Rewrite Alexa Smart-Home skill to v3 (#9699)

* Rewrite Alexa Smart-Home skill to v3

* add discovery & fix brigness

* Rewrite Tests

* fix lint

* fix lint p2

* fix version

* fix tests

* fix test message generator

* Update smart_home.py

* fix test

* fix set bug

* fix list

* fix response name for discovery

* fix flucky tests

balloob committed:
Fix I/O in event loop by Arlo alarm control panel (#9738)

tchellomello committed:
Make Arlo battery_level icon dynamic (#9747)

* Make Arlo battery_level icon dynamic

* makes lint happy

snjoetw committed:
Fix for TypeError in synology camera (#9754)

rytilahti committed:
switch.tplink, light.tplink: bump the pyhs100 version and adapt to api changes (#9454)

* bump the pyhs100 version and fix api changes

* switch.tplink: avoid I/O during __init__

* initialize _name to None in __init__

* update requirements_all.txt for the new version

balloob committed:
RFC: Use bind_hass for helpers (#9745)

* Add Helpers bind_hass functionality

* Update other helpers

MisterWil committed:
Skybell (#9681)

* New Skybell platform with components

* Added skybell components to omit.

* Preemptively fixing lint issues (hopefully).

* Removed unused variable.

* Requested changes.

* Additional CRs

* Hopefully the last of the CR's!

rytilahti committed:
add myself to yeelight owners, too (#9759)

Danielhiversen committed:
Update CODEOWNERS (#9760)

bachya committed:
Adding my contributions (#9761)

ChristianKuehnel committed:
Initializing statistics sensor with data from database (#9753)

* Initializing statistics sensor with data from database

* fixed broken test case

* usage of recorder component is now optional, thx to @andrey-git

* added test case for initialize_from_database

andrey-git committed:
Match test requirements by full package name. (#9764)

rytilahti committed:
yeelight: implement min_mireds and max_mireds, fixes #9509 (#9763)

* yeelight: implement min_mireds and max_mireds, fixes #9509

thanks to @amelchio for pointing this out!

* remove typing infos

tchellomello committed:
Bump raincloudy version 0.0.3 (#9767)

* Bump raincloudy version 0.0.3

* Fix logic for raincloudy status binary_sensor

* Simplified binary_sensor logic

* Simplify

syssi committed:
Xiaomi Smart WiFi Socket and Smart Power Strip integration (#9138)

* Xiaomi Smart WiFi Socket and Smart Power Strip integration

* Comment updated.

* Blank line removed.

* Typo fixed.

* Version of python-mirobo bumped.

* Version of python-mirobo bumped: Lightweight API changes.

* Additional API changes.

* Library version properly pinned again.

* Platform not ready behavior fixed.
Expose the device model as sensor attribute.
Device initialized log message added. Provides device model, firmware and hardware version.

* Component renamed: switch.xiaomi_plug -> switch.xiaomi_miio

* Revise based on review: Unused code removed. Filename updated.

etsinko committed:
fix for LocationParseError in netgear platform (#9683)

* fix for LocationParseError in netgear platform

* added unit tests for get_scanner()

* fixed houndci-bot warnings

* fixed lint warnings

* fixed lint warnings

* fixed broken test

* removed guard clause from netgear.py
removed all discovery related code from device_tracker
removed unnecessary unit test

* removed discovery related tests

* removed unused import

* removed unused import

balloob committed:
Expose time module in Python Scripts (#9736)

* Expose time module in Python Scripts

* Make dt_util available in Python Scripts

* Limit methods in time module

* Add time.mktime

* Limit access to datetime

* Add warning to time.sleep

* Lint

webworxshop committed:
Add notification platform for Rocket.Chat. (#9553)

* Add notification platform for Rocket.Chat.

* Changes to Rocket.Chat notification platform based on feedback.

* Implement better error handling for Rocket.Chat platform.

* Return None if Rocket.Chat notify platform init fails.

* Refactor Rocket.Chat notifications.

Refactor Rocket.Chat notification platform to remove async and
simplify error handling.

* fix url

zabuldon committed:
Merge branch 'dev' of https://github.com/zabuldon/home-assistant into dev

zabuldon committed:
Dep version bump. Remove unused assignment.
